DSCResources/VE_SFStoreRegisterGateway/VE_SFStoreRegisterGateway.ps1

```diff
@@ -0,0 +1,216 @@
+        }
+
+        ## Only check all remaing properties if we're setting
+        if ($Ensure -eq 'Present') {
+            if(($targetResource.Gateway -eq $Gateway) -and  ($targetResource.DefaultGateway -eq $DefaultGateway) -and  ($targetResource.UseFullVpn -eq $UseFullVpn)) {
+                    $inDesiredState = $true;
+            }
+            else {
+                $inDesiredState = $false;
+            }
+        }
+
+        if ($inDesiredState) {
+
+            Write-Verbose ($localizedData.ResourceInDesiredState);
+            return $true;
+        }
+        else {
+
+            Write-Verbose ($localizedData.ResourceNotInDesiredState);
+            return $false;
+        }
+
+    } #end process
+} #end function Test-TargetResource
+
+
+function Set-TargetResource {
+    [CmdletBinding()]
+
+    param (
+        ## Citrix Storefront Store Service IIS Virtual Path
+        [Parameter(Mandatory)]
+        [System.String] $StoreVirtualPath,
+
+
+        ## Citrix Storefront Store Service IIS Site Id
+        [Parameter()] [ValidateNotNull()]
+        [System.UInt16] $SiteId = 1,
+
+
+        ## Netscaler Gateway to register
+        [Parameter(Mandatory)] [ValidateNotNull()]
+        [System.String] $Gateway,
+
+
+        ## Use this Gateway as the default if more than one is defined
+        [Parameter()] [ValidateNotNull()]
+        [System.Boolean] $DefaultGateway = $true,
+
+
+        ## Use full VPN access when accessing the Store through this Gateway
+        [Parameter()] [ValidateNotNull()]
+        [System.Boolean] $UseFullVpn = $false,
+
+        [Parameter()] [ValidateSet('Present','Absent')]
+        [System.String] $Ensure = 'Present'
+    )
+    
+    process {
+
+        ImportSFModule -Name 'Citrix.StoreFront.Stores';
+        ImportSFModule -Name Citrix.Storefront.Authentication;
+
+        Write-Verbose ($localizedData.UpdatingStoreRegisteredGateway -f $StoreVirtualPath);
+
+        $targetResource = Get-TargetResource @PSBoundParameters;
+        
+        #Get the store and gateway objects from Storefront
+        $GatewayObject = Get-STFRoamingGateway -Name $Gateway;
+        $storeService = Get-STFStoreService -VirtualPath $StoreVirtualPath;
+
+        #
+        if ($Ensure -eq 'Absent') {
+            if($targetResource.Gateway -eq $Gateway) {
+                ## Netscaler Gateway registered to store, removing
+                [ref] $null = Unregister-STFStoreGateway -Gateway $GatewayObject -StoreService $storeService;
+            }
+        }
+        elseif ($Ensure -eq 'Present') {
+            #The CitrixAGBasic authentication protocol is used when accessing StoreFront remotely through a Citrix NetScaler Gateway
+            $AuthenticationService = Get-STFAuthenticationService -VirtualPath $storeService.AuthenticationServiceVirtualPath;
+            [ref] $null = Add-STFAuthenticationServiceProtocol -AuthenticationService $authenticationService -Name 'CitrixAGBasic';

```

Comment:
I confirm that it will not throw an error if the 'CitrixAGBasic' method is already present. Tested by adding a second gateway to a store already configured with a Netscaler Gateway and CitrixAgBasic authentication